org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/markers/MarkerUtils.java

```diff
@@ -66,6 +72,25 @@ public static void updateAllSonarMarkerSeverity() throws CoreException {
       }
     }
   }
+  
+  /**
+   *  Get the connection mode of a specific marker by id
+   *  
+   *  @param markerId for the marker <-> project connection
+   *  @param markerServerKey marker information from the connection, null if not on server
+   *  @return markers' corresponding project connection mode
+   */
+  public static String getProjectConnectionMode(IMarker marker, @Nullable String markerServerKey) {
+    var bindingOptional = SonarLintCorePlugin.getServersManager()
+      .resolveBinding(Adapters.adapt(marker.getResource().getProject(), ISonarLintProject.class));
+    if (bindingOptional.isEmpty() || markerServerKey == null) {
+      return SONAR_MARKER_MODE_STANDALONE;
```

Comment:
This is a bit confusing to have "standalone" when the project is actually bound. Maybe the names are not correct. We want to know if the issue is matched with a server issue.
So maybe have an enum with values:
```
enum FindingMatchingStatus {
  NOT_MATCHED,
  MATCHED_WITH_SQ,
  MATCHED_WITH_SC
}
```

Reply:
Done